internal/pkg/grpc/client.go

```diff
@@ -49,13 +48,14 @@ type eventHandler struct {
 }
 
 // NewClient returns a gRPC client.
-func NewClient(host string, insecure bool, timeoutSeconds int) Client {
-	return Client{host: host, timeoutSeconds: timeoutSeconds, insecure: insecure, connClose: func() error { return nil }}
+func NewClient(host string, insecure bool) Client {
+	return Client{host: host, insecure: insecure, connClose: func() error { return nil }}
 }
 
 // Connect attempts to establish a connection with a gRPC server.
 func (c *Client) Connect(headers []string) error {
-	ctx, cancel := context.WithTimeout(context.Background(), time.Duration(c.timeoutSeconds)*time.Second)
+	timeoutSeconds := 30 // TODO: make this configurable?
```

Comment:
same as readiness time?

Reply:
Hmm. This is slightly different.
We can have an app with a quick startup time but slow responses. In this case we would be forcing the user to set a higher than needed readiness time, otherwise the warmup requests would fail.

But as a temporary solution, I guess we can use the readiness and then come up with a better solution.

Reply:
Actually, this is used as a connect timeout and not read timeout.